sql-plugin/src/main/scala/org/apache/spark/sql/rapids/mathExpressions.scala

```diff
@@ -419,6 +419,81 @@ case class GpuCot(child: Expression) extends GpuUnaryMathExpression("COT") {
   }
 }
 
+case class GpuHypot(left: Expression, right: Expression) extends CudfBinaryMathExpression("HYPOT") {
+
+  override def binaryOp: BinaryOp = BinaryOp.ADD
+
+  // naive implementation of hypot
+  // r = sqrt(lhs^2 + rhs^2) 
+  // This is prone to overflow with regards either square term
+  // However, we can reduce it to
+  // r = sqrt(x^2 + y^2) = x * sqrt(1 + (y/x)^2)
+  // where x = max(abs(lhs), abs(rhs)), y = min(abs(lhs), abs(rhs))
+  // which will only overflow if both terms are near the maximum representation space
+  override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
+    // Using this form, because we really need a max and min that aren't NULL-aware
+    // In that case, then result will always be NULL, which is consistent with Spark
+    val (x, y) = withResource(Seq(lhs.getBase.abs, rhs.getBase.abs)) { 
+      case Seq(lhsAbs, rhsAbs) => {
+        withResource(lhsAbs.greaterThan(rhsAbs)) { lhsGreater => 
+          (lhsGreater.ifElse(lhsAbs, rhsAbs), lhsGreater.ifElse(rhsAbs, lhsAbs))
```

Comment:
nit: We are doing two computations here, and in a number of other places too. If the second computation threw an exception the result of the first one would leak.  It would be nice to clean it up a bit. Something like.

```scala
withResource(lhsAbs.greaterThan(rhsAbs)) { lhsGreater => 
    closeOnExcept(lhsGreater.ifElse(lhsAbs, rhsAbs)) { => x
        (x, lhsGreater.ifElse(rhsAbs, lhsAbs))
    }
}
```